lte/gateway/python/magma/pipelined/app/gw_mac_address.py

```diff
@@ -0,0 +1,124 @@
+"""
+Copyright 2022 The Magma Authors.
+
+This source code is licensed under the BSD-style license found in the
+LICENSE file in the root directory of this source tree.
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+import ipaddress
+import logging
+from typing import Optional
+
+from lte.protos.mobilityd_pb2 import IPAddress
+from scapy.arch import get_if_addr, get_if_hwaddr
+from scapy.data import ETH_P_ALL, ETHER_BROADCAST
+from scapy.error import Scapy_Exception
+from scapy.layers.inet6 import getmacbyip6
+from scapy.layers.l2 import ARP, Dot1Q, Ether
+from scapy.sendrecv import srp1
+
+EGRESS = "egress"
+
+
+def get_gw_mac_address(ip: IPAddress, vlan: str, non_nat_arp_egress_port: Optional[str]) -> str:
+    if ip.version == IPAddress.IPV4:
+        return _get_gw_mac_address_v4(ip, vlan, non_nat_arp_egress_port)
+    if ip.version == IPAddress.IPV6:
+        if vlan == "NO_VLAN":
+            return _get_gw_mac_address_v6(ip)
+        else:
+            gw_ip = ipaddress.ip_address(ip.address)
+            logging.error("Not supported: GW IPv6: %s over vlan %d", str(gw_ip), vlan)
+            return ""
+
+
+def _get_gw_mac_address_v4(ip: IPAddress, vlan: str, non_nat_arp_egress_port: Optional[str]) -> str:
```

Comment:
quick look into scapy: `non_nat_arp_egress_port` does not seem to be optional/none-safe. Does it make sense to have the ipv4/6 switch in egress.py?